sale_stock_line_customer_ref/views/stock_move_line.xml

```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2022 Camptocamp SA
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+<odoo>
+    <record id="view_move_line_form" model="ir.ui.view">
+        <field name="name">stock.move.line.form.inherit</field>
+        <field name="model">stock.move.line</field>
+        <field name="inherit_id" ref="stock.view_move_line_form" />
+        <field name="arch" type="xml">
+            <field name="product_id" position="after">
+                <field name="customer_ref" invisible="not customer_ref" />
+            </field>
+        </field>
+    </record>
+
+    <record id="view_stock_move_line_detailed_operation_tree" model="ir.ui.view">
+        <field name="name">stock.move.line.operations.list.inherit</field>
+        <field name="model">stock.move.line</field>
+        <field
+            name="inherit_id"
+            ref="stock.view_stock_move_line_detailed_operation_tree"
+        />
+        <field name="arch" type="xml">
+            <field name="product_id" position="after">
+                <field
+                    name="customer_ref"
+                    column_invisible="not parent.has_customer_ref"
```

Comment:
Caused by: EvalError: Can not evaluate python expression: (bool(not parent.has_customer_ref))

Reply:
thanks for the in4. from odoo 17.0, moves detailed operation view to smart button (see https://github.com/odoo/odoo/pull/140898/commits/2c1881dd) mode instead since `move_line_nosuggest_ids` is no longer used, so i added override `action_detailed_operations` function to be able to fix the above error